_posts/2024-04-29-dependabot-management.markdown

```diff
@@ -0,0 +1,206 @@
+
+          gh secret set AWS_CODEARTIFACT_TOKEN --app dependabot --org ${{ env.ORG }} --visibility all < aws-codeartifact-token.txt
+```
+
+<details>
+<summary>
+SEE FULL CODE HERE
+</summary>
+
+```yaml
+name: Manage AWS CodeArtifact Secret
+
+# This GitHub workflow, automates the process of creating, or updating a secret at the organization level using a GitHub App.
+
+on:
+push:
+ branches:
+   - main
+schedule:
+ - cron: "50 */7 * * *" # At minute 50 past every 7th hour every day.
+
+jobs:
+manage-secret:
+ runs-on: ubuntu-latest
+ environment: dev
+
+ permissions:
+   id-token: write
+   contents: read
+
+ steps:
+   - uses: actions/checkout@v3
+
+   - name: Generate GitHub App Token
+     id: generate_token
+     uses: tibdex/github-app-token@v1
+     with:
+       app_id: ${{ secrets.APP_ID }}
+       installation_id: ${{ secrets.INSTALLATION_ID }}
+       private_key: ${{ secrets.PRIVATE_KEY }}
+
+   - name: Assume deployment role
+      uses: aws-actions/configure-aws-credentials@v3
+     with:
+       role-to-assume: arn:aws:iam::1234567890:role/example-role
+       role-session-name: samplerolesession
+       aws-region: ap-southeast-2
+
+   - name: Set AWS CodeArtifact Secrets for Actions and Dependabot
+     env:
+       GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
+       ORG: *YOUR_ORG_NAME*
+     run: |
+       aws codeartifact get-authorization-token --domain *ANONYMOUS* --domain-owner *ANONYMOUS* --query authorizationToken --output text --region ap-southeast-2 --duration-seconds 28800 > aws-codeartifact-token.txt
+
+       gh secret set AWS_CODEARTIFACT_TOKEN --app dependabot --org ${{ env.ORG }} --visibility all < aws-codeartifact-token.txt
+```
+
+</details>
+
+<br>
+
+## Transient Dependencies
+
+One final gotcha I found while working with Dependabot is transient dependencies or dependencies of dependencies. Specifically, when you want combine a large number of PRs each containing a separate transient dependency. There is no special way to do this, and in fact the best way for node packages in `npm upgrade`. A more blunt and arguably stupid way to do it is to delete the `package-lock.json` file and then run `npm install`. This will update all the dependencies to the latest version, however, this is not recommended as it defeats the purpose of having a lock file in the first place, being a source of truth for the dependencies of the application.
+
+## A Note on Testing
+
+Updating packages can be painful for a number of reasons. One of particular importance is ensuring nothing breaks once something is updated. If your application has no testing, and you wish for your application to continue working after complete a package update, you need testing in place. If this testing is manual, tackling 49 pull requests can take a long time. With automated testing, the time taken can be vastly reduced. If you want exemplary automated unit, integration and end to end testing, I recommend chatting to Mechanical Rock.
```

Comment:
perhaps hyper link to Mechanical Rock contact page on website..

Reply:
done